tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,146 @@
+			NetworkSource: kvirtv1.NetworkSource{
+				Multus: &kvirtv1.MultusNetwork{
+					NetworkName: secondaryLinuxBridgeInterface.Name,
+				},
+			},
+		}
+
+		BeforeEach(func() {
+			By("Create NetworkAttachmentDefinition")
+			nad := generateNetworkAttachmentDefinition()
+			_, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(util.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			defaultBridgeInterface := libvmi.InterfaceDeviceWithBridgeBinding(defaultInterfaceName)
+			vmiSpec := libvmi.NewFedora(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&defaultBridgeInterface, originalMac)),
+				libvmi.WithNetwork(kvirtv1.DefaultPodNetwork()),
+				libvmi.WithInterface(secondaryLinuxBridgeInterface),
+				libvmi.WithNetwork(&linuxBridgeNetwork),
+				libvmi.WithCloudInitNoCloudUserData("#!/bin/bash\nsudo ip link set dev eth0 address "+updatedMac+"\n", false))
+
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmiSpec)
+			Expect(err).NotTo(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+			tests.WaitAgentConnected(virtClient, vmi)
+		})
+
+		It("should have the expected entries in vmi status when mac is changed", func() {
+			expectedInterfaces := []kvirtv1.VirtualMachineInstanceNetworkInterface{
+				{
+					InfoSource: "domain",
+					MAC:        originalMac,
+					Name:       defaultInterfaceName,
+				},
+				{
+					InfoSource:    "domain, guest-agent",
+					InterfaceName: "eth1",
+					MAC:           secondaryLinuxBridgeInterface.MacAddress,
+					Name:          secondaryLinuxBridgeInterface.Name,
+				},
+				{
+					InfoSource:    "guest-agent",
+					InterfaceName: "eth0",
+					MAC:           updatedMac,
+				},
+			}
+
+			Eventually(func() error {
+				vmiObj, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				if len(vmiObj.Status.Interfaces) != len(expectedInterfaces) {
+					return fmt.Errorf("interfaces doesn't match len error")
+				}
+
+				for i := range vmiObj.Status.Interfaces {
+					expectedInterfaces[i].IP = vmiObj.Status.Interfaces[i].IP
+					expectedInterfaces[i].IPs = vmiObj.Status.Interfaces[i].IPs
+				}
+
+				if reflect.DeepEqual(expectedInterfaces, vmiObj.Status.Interfaces) {
+					return nil
+				}
+				return fmt.Errorf("interfaces doesn't match\n%v\n%v", expectedInterfaces, vmiObj.Status.Interfaces)
+			}, 140*time.Second, 2*time.Second).Should(Succeed())
+		})
+	})
+})
+
+func generateNetworkAttachmentDefinition() *k8snetworkplumbingwgv1.NetworkAttachmentDefinition {
+	return &k8snetworkplumbingwgv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "secondary",
+		},
+		Spec: k8snetworkplumbingwgv1.NetworkAttachmentDefinitionSpec{Config: `{"cniVersion": "0.3.1", "name": "secondary", "type": "cnv-bridge", "bridge": "br1"}`},
```

Comment:
Same here, the `br1` is too generic, try to give it a more specialized name of you need an unique bridge/nad.

If there is no need for an unique bridge/nad, then consider showing that you share the same one that is already used under multus tests.
Also, please make sure we can run such tests in parallel, as it does use a common resource on the node. Not sure if it is a problem or not, it just needs to be considered well so we will not cause collisions.

Reply:
> Also, please make sure we can run such tests in parallel, as it does use a common resource on the node. Not sure if it is a problem or not, it just needs to be considered well so we will not cause collisions.

It is not a problem (and was tested).
It is created in a namespace which is parallel safe,
each parallel instance creates its own namespaces
```
			_, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(util.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
```

I think the code here has only what we need, and the creation flow is simpler than
https://github.com/kubevirt/kubevirt/blob/77193f17822e4e3a781c116d340a7cbb7e42c6ce/tests/network/vmi_multus.go#L162
which uses POST, and more unneeded variables.

so i prefer to use this,
we can consider a follow PR that will create lib for NADs and will be used on both places
but for now the code on multus is more complex than what we need

Reply:
> It is not a problem (and was tested).
> It is created in a namespace which is parallel safe,
> each parallel instance creates its own namespaces, and then we use it as the NAD ns

I do not know how you are able to be sure it is not a problem when running in parallel. It is not like you control which tests run at the same time. I cannot think of a reasonable way to control this.

I pointed out the bridge created on the node and you are talking about the NAD which can be consumed by pods.
I think you miss the part where this CNI bridge creates in a lazy manner a linux-bridge on the node. That part is a shared resource and has nothing to do with the namespace. If I am wrong, please correct me.

> so i prefer to use this, as its clean and compact (and uses the basic high level API which suitable here)

I do not mind you using the structured version, I do mind that two NAD/s point to the same underlying bridge, that sounds problematic to me.

Reply:
> I do not know how you are able to be sure it is not a problem when running in parallel. It is not like you control which tests run at the same time. I cannot think of a reasonable way to control this.

In aspect NADs only, we do control the namespaces, each parallel instance has its own namespaces.

about the underlying bridge that is created on the node (rest of the comment), i changed the name of the bridge to `infosrc`, would it be enough ?